daos/ScoreDAO.js

```diff
@@ -354,7 +358,7 @@ class ScoreDAO {
 			`
 				SELECT id, datetime, start_level, end_level, score, lines, tetris_rate, num_droughts, max_drought, das_avg, duration, frame_file, competition
 				FROM scores
-				WHERE player_id=$1 ${filter_by_competition_mode}
+				WHERE player_id=$1 ${filter_by_competition_mode} ${options.level ? `AND start_level=${options.level}` : ''}
```

Comment:
Looking at this again, actually I reckon the code here should match `getNumberOfScores()`:
1. `filter_by_competition_mode` should be renamed to `additional_conditions`
2. the extra conditions should be added in parametrized query style (prep `$n` placeholders, and populate the `args` array accordingly)

For reference, yet another function that employ this pattern in this file is [`getProgress()`](https://github.com/nestrischamps/nestrischamps/blob/08c52cd35413b0fa8f43e87bc815546fb5cefd81/daos/ScoreDAO.js#L403-L436)